src/orquestra/quantum/wavefunction.py

```diff
@@ -214,12 +215,16 @@ def flip_wavefunction(wavefunction: Wavefunction):
 
 
 def flip_amplitudes(amplitudes: Union[Sequence[complex], np.ndarray]) -> np.ndarray:
-    number_of_states = len(amplitudes)
-    ordering = [
+    ordering = _get_ordering(len(amplitudes))
+    return np.array([amplitudes[i] for i in ordering])
+
+
+@lru_cache(maxsize=8)
```

Comment:
[question] Why 8?

Reply:
The `ordering` variable can be large so I didn't want to cache too many of them. As far as I could tell it's traditional to use powers of two. So I settled on 8. I could boost it or make it smaller if you want.

Reply:
I think that storing amplitudes shouldn't be a memory bottleneck, as for wavefunction of size N, your circuit is roughly N^2, so if anything, you'll blow up there.
And cache of size 8 looks a little bit small to me, especially if you have a 20-qubit wavefunction with 1 million entries, and that's when this would start being really useful I guess. So it looks like it would be worth increasing?

Feel free to disagree!

Reply:
@mstechly but the wavefunction itself (I mean, its amplitudes) is irrelevant, as the caching is based on lengths. So, the number of different lengths of wave functions is what matters. And unless we are planning to have lots of wavefunctions with different lengths, I think this cache size should be sufficient.

Having said that, it would be nice to see some profiling evidence demonstrating the usefulness of introducing cache here :)

Reply:
Sure! I'll increase the size!